includes/admin/class-wc-stripe-settings-controller.php

```diff
@@ -68,13 +68,17 @@ public function set_stripe_gateways_in_list( $ordering ) {
 	* @param WC_Order $order The order that is being viewed.
 	*/
 	public function hide_refund_button_for_uncaptured_orders( $order ) {
-		$intent = $this->gateway->get_intent_from_order( $order );
-
-		if ( $intent && 'requires_capture' === $intent->status ) {
-			$no_refunds_button  = __( 'Refunding unavailable', 'woocommerce-gateway-stripe' );
-			$no_refunds_tooltip = __( 'Refunding via Stripe is unavailable because funds have not been captured for this order. Process order to take payment, or cancel to remove the pre-authorization.', 'woocommerce-gateway-stripe' );
-			echo '<style>.button.refund-items { display: none; }</style>';
-			echo '<span class="button button-disabled">' . $no_refunds_button . wc_help_tip( $no_refunds_tooltip ) . '</span>';
+		try {
+			$intent = $this->gateway->get_intent_from_order( $order );
+
+			if ( $intent && 'requires_capture' === $intent->status ) {
+				$no_refunds_button  = __( 'Refunding unavailable', 'woocommerce-gateway-stripe' );
+				$no_refunds_tooltip = __( 'Refunding via Stripe is unavailable because funds have not been captured for this order. Process order to take payment, or cancel to remove the pre-authorization.', 'woocommerce-gateway-stripe' );
+				echo '<style>.button.refund-items { display: none; }</style>';
+				echo '<span class="button button-disabled">' . $no_refunds_button . wc_help_tip( $no_refunds_tooltip ) . '</span>';
+			}
+		} catch ( WC_Stripe_Exception $e ) {
```

Comment:
```suggestion
		} catch ( Exception $e ) {
```

I believe it needs to be a generic exception rather than `WC_Stripe_Exception`. 

Following the steps to test, adding a `throw new Exception( ... )` didn't get caught.

Looking at `get_intent_from_order()` and the subsequent `get_intent()`, it can throw a generic exception if the type is invalid ([here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/fix/3329-uncaught-error/includes/abstracts/abstract-wc-stripe-payment-gateway.php#L1633)) or the API could throw a `WC_Stripe_Exception` if the response is invalid so the more generic exception type will catch both of those.

Reply:
You are right @james-allan. I had already made this change but forgot to push the commit 🤦‍♀️